src/renderers/shared/fiber/ReactFiberUpdateQueue.js

```diff
@@ -28,6 +28,36 @@ const {
 const validateCallback = require('validateCallback');
 const warning = require('warning');
 
+// This is a generic linked-list type. We're using it for callbacks here but
+// it could be reused for other purposes.
+type ListNode<T> = {
+  value: T,
+  next: ListNode<T> | null,
+};
+
+type List<T> = {
+  first: ListNode<T> | null,
+  last: ListNode<T> | null,
+};
+
+function createList<A>() : List<A> {
+  return { first: null, last: null };
```

Comment:
Could also make the List type an intersection of the ListNode type, like the UpdateQueue type was in the first version.